### PR TITLE
[v7r1] bdii2csagent purgeSites fix

### PR DESCRIPTION
**Pull request by andresailer**
@fstagni this is currently the hotfix on lbcertifdirac, so the last email you received should only contain "new" cern CEs.

Did you also want to receive notification that there are sites ignored? Because that could result in emails being sent every time.

BEGINRELEASENOTES

*CS
FIX: Bdii2CSAgent: Fix exception if selectedSites was set, but some unknown Sites were found by the agent
CHANGE: Bdii2CSAgent: if selectedSites is configured, also remove all CEs from unknown sites from the results

*Framework
CHANGE: NotificationHandler: sendMail: the `avoidSpam` parameter is deprecated. All emails (same subject, address, body) are now only sent once per 24h

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
There was a problem with the merge for the NotificationHandler.

There are also two logics for spam avoidance in the Handler, one enabled with avoidSpam=True caching ~1hour, and one active for avoidSpam=False, caching emails for 24 hours. I think we can maybe merge those two?

File: ConfigurationSystem/Agent/Bdii2CSAgent.py [hunk 1]
```diff
@@ -304,11 +305,16 @@ def __purgeSites(self, ceBdiiDict):
...
         self.log.error("No CE information for site:", site)
         continue
-      res = getCESiteMapping(ces[0])
-      if not res['OK']:
-        self.log.error("Failed to get DIRAC site name for ce", "%s: %s" % (ces[0], res['Message']))
-        continue
-      siteInCS = res['Value'][ces[0]]
+      siteInCS = 'Not_In_CS'
+      for ce in ces:
+        res = getCESiteMapping(ce)
```
Comment (fstagni) [inline ConfigurationSystem/Agent/Bdii2CSAgent.py:310]: Calling `getCESiteMapping()` (without input `ce`) would get you the whole mapping. You can call it outside of the loop.

**Comment by fstagni** [comment]
- I don't want to receive notifications for sites that are ignored, but I'd like to receive notifications for new sites, if there are.
- please merge the 2 things for avoiding spam

**Comment by andresailer** [comment]
> I don't want to receive notifications for sites that are ignored, but I'd like to receive notifications for new sites, if there are.

This is kind of how things were before.

So: if "SelectedSites" is set (like now in the certification), do you want to receive notifications for CEs from other sites that are not yet part of the CS?

**Comment by fstagni** [comment]
> > I don't want to receive notifications for sites that are ignored, but I'd like to receive notifications for new sites, if there are.
> 
> This is kind of how things were before.
> 
> So: if "SelectedSites" is set (like now in the certification), do you want to receive notifications for CEs from other sites that are not yet part of the CS?

Mmm I see your point now. Yes, I guess I want to receive notifications for CEs from other sites that are not yet part of the CS. It will be a pain for certification but for production is the way to go.

**Comment by andresailer** [comment]
We can also leave this implementation. In production you probably wouldn't set SelectedSites, unless you run another Bdii2Csagent for other sites, which is what I did for a while, when I used glue2 for CERN and glue1 for the rest.

For avoidspam: use 24 hour caching for all emails? I like the DictCache implementation better than the every hour purge.

**Comment by fstagni** [comment]
I would not set "SelectedSites" in production.

OK for 24 hour caching.

**Comment by andresailer** [comment]
I removed the avoidSpam/gMailSet logic.

**Review by fstagni — APPROVED** [review]

**Review by atsareg — APPROVED** [review]

**Pull request merged**